File: rev-com-server/src/main/java/com/codergb/bean/movie/Movie.java
package com.codergb.bean.movie;

import com.codergb.bean.FileInfo;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import java.util.Date;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(callSuper = true)
public class Movie  extends FileInfo {
  private String id;
  private String name;
  private float score;
  private List<Director> directors;
  private List<Screenwriter> screenwriters;
  private String coverUrl;
  private List<Area> area;
  private String language;
  private Date releaseTime;
  private Long duration;
  private String alias;
  private Form form;
  private Integer comment;
  private List<Actor> actors;
  private List<Category> categories;
  private String description;
}

File: rev-com-client/src/types/movie/movie.ts
import {IActor} from "./actor";
import {ICategory} from "./ICategory";
import {IDirector} from "./director";
import IArea from "./IArea";
import {IForm} from "./IForm";
import {IScreenwriter} from "./screenwriter";

interface IMovie{
  id: string
  name: string
  directors: IDirector[]
  screenwriters: IScreenwriter[]
  coverUrl: string,
  score:number,
  area: IArea[]
  language: string
  releaseTime: string
  duration: number
  alias: string
  form: IForm,
  comment:number,
  actors: IActor[]
  categories: ICategory[]
  description: string
}
export type{
  IMovie
}
